tests/unit/src/test/scala/tests/decorations/SyntheticDecorationsLspSuite.scala

```diff
@@ -197,7 +210,68 @@ class SyntheticDecorationsLspSuite extends BaseLspSuite("implicits") {
            |  }
            |  implicit val andy : String = "Andy"
            |  hello()
-           |  "foo".map[Char, String](c => c.toUpper)
+           |  ("1" + "2").map[Char, String](c => c.toUpper)
+           |}
+           |""".stripMargin
+      )
+      _ <- server.didChangeConfiguration(
+        """{
+          |  "show-implicit-arguments": false,
+          |  "show-implicit-conversions": true,
+          |  "show-inferred-type": false
+          |}
+          |""".stripMargin
+      )
+      _ = client.decorations.clear()
+      _ <- server.didOpen("a/src/main/scala/Main.scala")
+      _ = assertNoDiff(
+        client.workspaceDecorations,
+        """|object Main{
+           |  def hello()(implicit name: String) = {
+           |    println(s"Hello $name!")
+           |  }
+           |  implicit val andy : String = "Andy"
+           |  hello()
+           |  (augmentString("1" + "2")).map(c => charWrapper(c).toUpper)
+           |}
+           |""".stripMargin
+      )
+    } yield ()
+  }
+
+  test("augment-string") {
+    for {
+      _ <- server.initialize(
+        s"""|/metals.json
+            |{
+            |  "a": {}
+            |}
+            |
+            |/a/src/main/scala/Main.scala
+            |object Main{
+            |  ("1" + "2")
+            |    .stripSuffix(".")
+            |    .stripSuffix("#")
+            |    .stripPrefix("_empty_.")
+            |}
+            |""".stripMargin
+      )
+      _ <- server.didChangeConfiguration(
+        """|{
+           |  "show-implicit-conversions": true
+           |}
+           |""".stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/Main.scala")
+      _ <- server.didSave("a/src/main/scala/Main.scala")(identity)
+      _ = assertNoDiagnostics()
+      _ = assertNoDiff(
+        client.workspaceDecorations,
+        """|object Main{
+           |  (augmentString(augmentString(augmentString("1" + "2"))
```

Comment:
So this is a curious sample and I think it makes sense for to add some filters later on if people complain.

Looking through metals codebase it doesn't seem too bothersome, but might be be in some cases.

Reply:
lol yea this example isn't awesome, at the same time, I think it's fine. If people do start to complain about it, we can think about fine tuning it.